x/ibc/05-port/keeper/keeper.go

```diff
@@ -5,22 +5,23 @@ import (
 
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/capability"
 	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
 )
 
 // Keeper defines the IBC connection keeper
 type Keeper struct {
```

Comment:
How would we implement the `releasePort` function given the capabilityKeeper implementation? As far as I can tell, there's no way for a module to release ownership of a capability.

Also since modules and ports are 1-1 at any given point in time, not sure how to enforce this on the capability side since capabilities are multiowner. Perhaps this needs to be enforced elsewhere, maybe in the routing table?

Reply:
Yes, this needs to be added to ADR 3, good catch. 

The router will have to enforce 1-1 routing, it can do that with the existing capability API I think.